Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -0,0 +1,170 @@
+using Content.Server.Actions.Actions;
+using Content.Server.Lathe.Components;
+using Content.Server.Popups;
+using Content.Shared.Actions;
+using Content.Shared.Actions.Components;
+using Content.Shared.Damage;
+using Content.Shared.Interaction;
+using Content.Shared.MobState;
+using Content.Shared.MobState.Components;
+using Content.Shared.Popups;
+using JetBrains.Annotations;
+using Robust.Shared.Containers;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Localization;
+using Robust.Shared.Log;
+using Robust.Shared.Map;
+using System;
+
+namespace Content.Server.Guardian
+{
+    public class GuardianSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<GuardianCreatorComponent, UseInHandEvent>(OnGuardianCreated);
+            SubscribeLocalEvent<GuardianComponent, MoveEvent>(OnGuardianMove);
+            SubscribeLocalEvent<GuardianHostComponent, MoveEvent>(OnGuardianHostMove);
+            SubscribeLocalEvent<GuardianComponent, DamageChangedEvent>(OnGuardianDamaged);
+            SubscribeLocalEvent<GuardianHostComponent, MobStateChangedMessage>(OnHostDeath);
+        }
+
+        /// <summary>
+        /// Triggers upon mobstate, to detect disintigration command upon host's death
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="component"></param>
+        /// <param name="args"></param>
+        private void OnHostDeath(EntityUid uid, GuardianHostComponent component, MobStateChangedMessage args)
+        {
+            //We only care if the host is dead, not in crit or incapacitated, as he still can support a holopara
+            //Dragging him away
+            if (component.Owner.GetComponent<MobStateComponent>().IsDead())
+            {
+                //Delete both entities to prevent revival, guardian first to avoid errors
+                //TODO: add disintigration. Proper. Current method leaves no items behind.
+                EntityManager.QueueDeleteEntity(component._hostedguardian);
+                EntityManager.QueueDeleteEntity(uid);
```

Comment:
what? it just straight up deletes the host too?

Reply:
should probably just gib them so at least their items drop

Reply:
Currently working on dat